cloudstack/service_offering_schema.go

```diff
@@ -0,0 +1,248 @@
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.RequiresReplace(),
+			},
+			Default: booldefault.StaticBool(false),
+		},
+		"host_tags": schema.StringAttribute{
+			Description: "The host tag for this service offering",
+			Optional:    true,
+		},
+		"id": schema.StringAttribute{
+			Description: "uuid of service offering",
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"is_volatile": schema.BoolAttribute{
+			Description: "Service offering is volatile",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.RequiresReplace(),
+			},
+			Default: booldefault.StaticBool(false),
+		},
+		"limit_cpu_use": schema.BoolAttribute{
+			Description: "Restrict the CPU usage to committed service offering",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.RequiresReplace(),
+			},
+			Default: booldefault.StaticBool(false),
+		},
+		"name": schema.StringAttribute{
+			Description: "The name of the service offering",
+			Required:    true,
+		},
+		"network_rate": schema.Int32Attribute{
+			Description: "Data transfer rate in megabits per second",
+			Optional:    true,
+		},
+		"offer_ha": schema.BoolAttribute{
+			Description: "The HA for the service offering",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.RequiresReplace(),
+			},
+			Default: booldefault.StaticBool(false),
+		},
+		"zone_ids": schema.SetAttribute{
+			Description: "The ID of the zone(s)",
+			Optional:    true,
+			ElementType: types.StringType,
+		},
+		"disk_offering": schema.SingleNestedAttribute{
+			Optional: true,
+			Attributes: map[string]schema.Attribute{
+				"cache_mode": schema.StringAttribute{
+					Description: "the cache mode to use for this disk offering. none, writeback or writethrough",
+					Required:    true,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.RequiresReplace(),
+					},
+				},
+				"disk_offering_strictness": schema.BoolAttribute{
+					Description: "True/False to indicate the strictness of the disk offering association with the compute offering. When set to true, override of disk offering is not allowed when VM is deployed and change disk offering is not allowed for the ROOT disk after the VM is deployed",
+					Required:    true,
+					PlanModifiers: []planmodifier.Bool{
+						boolplanmodifier.RequiresReplace(),
+					},
+				},
+				"provisioning_type": schema.StringAttribute{
+					Description: "provisioning type used to create volumes. Valid values are thin, sparse, fat.",
+					Required:    true,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.RequiresReplace(),
+					},
+				},
+				"root_disk_size": schema.Int64Attribute{
+					Description: "the Root disk size in GB.",
+					Required:    true,
```

Comment:
@poddm I believe this is what is needed to resolve the issue when trying to create offerings with not specifying a `root_disk_size` or needing it to be `0`.  I made this change locally and tested and it worked.

```suggestion
					Optional:    true,
```